macrobot/macrobot/handler.go

```diff
@@ -36,6 +36,12 @@ func NewHandler(stats *base.StatsRegistry, kbc *kbchat.API, debugConfig *base.Ch
 }
 
 func (h *Handler) HandleNewConv(conv chat1.ConvSummary) error {
+	// When we're put into a team conv, tell the team about the `create-for-channel` option.
+	if conv.Channel.MembersType == "team" && conv.IsDefaultConv {
```

Comment:
Will this trigger if a restricted bot is only added to a specific channel?

Reply:
i'll put an lru in for if we've advertised to a conv before instead